bal_tools/subgraph.py

```diff
@@ -30,6 +30,33 @@
     "fraxtal": "https://graph.data.aura.finance/subgraphs/name/aura-finance-fraxtal",
     "avalanche": "https://subgraph.satsuma-prod.com/cae76ab408ca/1xhub-ltd/aura-finance-avalanche/api",
 }
+# https://raw.githubusercontent.com/balancer/docs-v3/refs/heads/v3-outline/docs/data-and-analytics/data-and-analytics/subgraph.md
+VAULT_V3_SUBGRAPHS_BY_CHAIN = {
+    "arbitrum": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/Ad1cgTzScNmiDPSCeGYxgMU3YdRPrQXGkCZgpmPauauk",
+    "base": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/9b7UBHq8DXxrfGsYhAzF3jZn5mNRgZb5Ag18UL9GJ3cV",
+    "ethereum": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/4rixbLvpuBCwXTJSwyAzQgsLR8KprnyMfyCuXT8Fj5cd",
+    "gnosis": "https://gateway.thegraph.com/api/[api-key]/subgraphs/id/DDoABVc9xCRQwuXRq2QLZ6YLkjoFet74vnfncQDgJVo2",
+}
+VAULT_V3_SUBGRAPHS_BY_CHAIN_DEV = {
+    "arbitrum": "https://api.studio.thegraph.com/query/75376/balancer-v3-arbitrum/version/latest",
+    "base": "https://api.studio.thegraph.com/query/75376/balancer-v3-base/version/latest",
+    "mainnet": "https://api.studio.thegraph.com/query/75376/balancer-v3/version/latest",
```

Comment:
note discrepancy that prod dicts have use `ethereum` and the dev ones  use `mainnet`

Reply:
yea this is a mess in general in the balancer code base... for me mainnet means nothing; mainnet of what chain?

anyway this is fixed by using your programmatic approach